gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -490,33 +517,82 @@ var gpii = fluid.registerNamespace("gpii");
     };
 
     /**
+     * The lifecycleManager queue is used to hold the high-level actions that needs to happen,
+     * such as starting the login process, starting logout process, starting the update process.
+     * Each item in the queue should have the format: { func: <functionToCall>, args: <Arguments> }
+     * Where <functionToCall> is a function that returns a promise for when it's complete (including side-effects)
```

Comment:
The system will be easier to debug (as well as easier to add more intelligent processing in future) if we record the functions in some kind of string form rather than as raw function handles.
Possibilities are i) the entire IoC reference e.g. "{that}.start", which we then call fluid.expandOptions(funcRef, that) on, 
ii) just the member name, which we then call that[funcName] to get the function
iii) others